clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1102,6 +1135,9 @@ private Record nextFetchedRecord() {
                     bytesRead += record.sizeInBytes();
                     nextFetchOffset = record.offset() + 1;
                 }
+                // In some cases, the deserialization may have thrown an exception and the retry may succeed,
```

Comment:
The scenario I am thinking is a deserializer that needs a remote call that cannot be completed at the first time, but passed the second time. This is probably a corner case.